bin/ut-release.js

```diff
@@ -1,14 +1,72 @@
 }, function(err, result) {
     if (err) {
         throw err;
     } else {
-        require('../lib/exec')('npm', ['version', result.releaseType, '-m', '[ci-skip][ci skip] version incremented to %s']);
-        require('../lib/exec')('npm', ['publish']);
-    };
+        var releaseType = result.releaseType;
+        var currentVersion = packageJson.version;
+        var tokens = [];
+        if (process.env.gitlabSourceBranch) {
+            tokens = process.env.gitlabSourceBranch.split('/');
+        } else if (process.env.GIT_BRANCH) {
+            tokens = process.env.GIT_BRANCH.split('/').slice(1); // remove origin
+        } else {
+            throw new Error('Branch name could not be detected!');
+        }
+        var versionToRelease;
+        if (tokens.length === 2 && buildableBranches[tokens[0]] && tokens[1]) {
+            releaseType = buildableBranches[tokens[0]];
+            if (releaseType === 'preminor') {
+                var prereleaseTokens = semver.prerelease(currentVersion);
+                if (prereleaseTokens) {
+                    if (prereleaseTokens[0] !== tokens[1]) {
+                        throw new Error(`Release version mismatch: Version ${prereleaseTokens[0]} can't be released from version ${tokens[1]}`);
+                    }
+                    releaseType = 'prerelease';
+                } else {
+                    if (!semver.prerelease(currentVersion + '-' + tokens[1])) {
+                        throw new Error(`incorrect branch name: ${tokens.join('/')}! ${tokens[1]} MUST comprise only [0-9A-Za-z-]`);
+                    }
+                    versionToRelease = semver.inc(currentVersion, releaseType, tokens[1]);
+                }
+            }
+        } else if (tokens.length > 1 || (tokens.length === 1 && tokens[0] !== 'master')) {
+            throw new Error(`incorrect branch name: ${tokens.join('/')}! Allowed branch names: master, feat/*, fix/*`);
+        }
+        if (!versionToRelease) {
+            versionToRelease = semver.inc(currentVersion, releaseType);
+        }
+        return pkgVersions(packageJson.name).then(function(versions) {
+            var conflictingSemverDiff = releaseType.startsWith('pre') ? 'prerelease' : releaseType;
+            var conflictingVersions = Array.from(versions).filter((version) => {
+                if (
+                    semver.eq(versionToRelease, version) ||
+                    (semver.lte(versionToRelease, version) && semver.diff(versionToRelease, version) === conflictingSemverDiff)
+                ) {
+                    return true;
+                }
+                return false;
+            });
+            if (conflictingVersions.length) {
+                throw new Error(`${releaseType} version ${versionToRelease} coudn't be published! Conflicting versions: ${conflictingVersions.join(', ')}`);
+            }
+            exec('npm', ['version', releaseType, '-m', '[ci-skip][ci skip] version incremented to %s']);
```

Comment:
it should execute one of the following:
npm version patch
npm version preminor
npm version prerelease
... depending on the case
shouldn't it?

the versionToRelease is used only to check there are no conflicts with other versions that are already published.